### PR TITLE
Add Learn PWA: Part 18, Window management.

### LINKED ISSUE #6497: Learn PWA: 19. Windows

### PR DESCRIPTION
**Pull request by jpmedley**
Fixes #6497 

Changes proposed in this pull request:

-
-
-

When you're ready to submit your PR, don't forget to add the `$-presubmit` label.

### CODE REVIEW

**Comment by rachelandrew** [comment]
@jpmedley this one is failing to build due to an AVIF image being used.

File: src/site/content/en/learn/pwa/windows/index.md [hunk 1]
```diff
@@ -1,11 +1,252 @@
 ---
-title: Windows
+layout: post
```
Comment (rachelandrew) [inline src/site/content/en/learn/pwa/windows/index.md:2]: I don't think we use this here

File: src/site/content/en/learn/pwa/windows/index.md [hunk 2]
```diff
@@ -1,11 +1,252 @@
...
+
+### Browsing to other sites
+If you want to send the user to an external site that is out of the scope of your PWA, you can do so with a standard `<a href>` HTML element, using `location.href` or opening a new window on compatible platforms.
+
+Currently on all browsers, if your PWA is installed, when you browse to a URL that is out of the [scope of your manifest](/learn/pwa/web-app-manifest/#recommended-fields), the browser engine of your PWA will render an in-app browser within the context of your window.
+
+Some features of the in-app browsers are:
+
+* They appear on top of your content.
+* They have a static URL bar showing the current origin, the window's title, and a menu. Typically, they'reit's themed with the `theme_color` of your manifest.
```
Comment (rachelandrew) [inline src/site/content/en/learn/pwa/windows/index.md:61]:
```suggestion
* They have a static URL bar showing the current origin, the window's title, and a menu. Typically, it's themed with the `theme_color` of your manifest.
```

File: src/site/content/en/learn/pwa/windows/index.md [hunk 3]
```diff
@@ -1,11 +1,252 @@
...
+* From the contextual menu, you can open that URL in the browser.
+* Users can close the browser or go back.
+
+{% Img src="image/RK2djpBgopg9kzCyJbUSjhEGmnw1/0SsThiNz4GDkJy6OiTHj.png", alt="An in-app browser on a desktop PWA when browsing a URL that is out-of-scope", width="800", height="497" %}
```
Comment (rachelandrew) [inline src/site/content/en/learn/pwa/windows/index.md:65]:
```suggestion
{% Img src="image/RK2djpBgopg9kzCyJbUSjhEGmnw1/0SsThiNz4GDkJy6OiTHj.png", alt="An in-app browser on a desktop PWA when browsing a URL that is out-of-scope.", width="800", height="497" %}
```

File: src/site/content/en/learn/pwa/windows/index.md [hunk 4]
```diff
@@ -1,11 +1,252 @@
...
+
+{% Aside %}
+While the in-app browser is on the screen, your PWA is waiting in the background, as if another window is obscuring it.
+{% endAside %}
+
+{% Aside 'gotchas' %}
+On iOS and iPadOS, the in-app browser uses SafariViewController, a Safari-rendering engine isolated from your PWA storage. Therefore, if the user has a session in Safari, it won't appear in your PWA, but it will appear within an in-app browser in your PWA.
+{% endAside %}
+
+{% Img src="image/RK2djpBgopg9kzCyJbUSjhEGmnw1/Gr4qJUx6hAFMyPLPyfQY.png", alt="An in-app browser on an iPhone when browsing a URL that is out-of-scope within a standalone PWA", width="800", height="370" %}
```
Comment (rachelandrew) [inline src/site/content/en/learn/pwa/windows/index.md:75]:
```suggestion
{% Img src="image/RK2djpBgopg9kzCyJbUSjhEGmnw1/Gr4qJUx6hAFMyPLPyfQY.png", alt="An in-app browser on an iPhone when browsing a URL that is out-of-scope within a standalone PWA.", width="800", height="370" %}
```

File: src/site/content/en/learn/pwa/windows/index.md [hunk 5]
```diff
@@ -1,11 +1,252 @@
...
+
+{% Img src="image/RK2djpBgopg9kzCyJbUSjhEGmnw1/UyohW4R5Sy79ffHri8cz.png", alt="An in-app browser on Android when browsing a URL that is out-of-scope within a standalone PWA", width="800", height="1644" %}
```
Comment (rachelandrew) [inline src/site/content/en/learn/pwa/windows/index.md:77]:
```suggestion
{% Img src="image/RK2djpBgopg9kzCyJbUSjhEGmnw1/UyohW4R5Sy79ffHri8cz.png", alt="An in-app browser on Android when browsing a URL that is out-of-scope within a standalone PWA.", width="800", height="1644" %}
```

File: src/site/content/en/learn/pwa/windows/index.md [hunk 6]
```diff
@@ -1,11 +1,252 @@
...
+
+On desktop, users can open more than one window of the same PWA. Each window will be a different navigation to the same `start_url`, as if you were opening two browser tabs of the same URL.
+From the menu in the PWA, users can select File then New window, and from your PWA code, you can open a new window with the `open()` function. Check the [documentation](https://developer.mozilla.org/docs/Web/API/Window/open) for details.
+
+```js
+function openNewWindow() {
+	window.open("/", "new-window", "width=600,height=600");
+}
+```
+{% Img src="image/RK2djpBgopg9kzCyJbUSjhEGmnw1/CURym4RD66CqcTw4iPl5.png", alt="The same installed PWA with several windows opened on a desktop operating system", width="800", height="580" %}
```
Comment (rachelandrew) [inline src/site/content/en/learn/pwa/windows/index.md:114]:
```suggestion
{% Img src="image/RK2djpBgopg9kzCyJbUSjhEGmnw1/CURym4RD66CqcTw4iPl5.png", alt="The same installed PWA with several windows opened on a desktop operating system.", width="800", height="580" %}
```

File: src/site/content/en/learn/pwa/windows/index.md [hunk 7]
```diff
@@ -1,11 +1,252 @@
...
+{% Aside %}
+To reduce phishing within PWAs, some desktop browsers may have additional measures on the window's title. For example, if you don't use your app's name in the title, the browser may add a prefix with your PWA's name on it. Other browsers may render the current origin in the title bar for a few seconds when you change the title to highlight where the user navigated.
+{% endAside %}
+
+### Tabbed mode
+
+An experimental capability, known as *tabbed mode*​ will let your PWA have a tab-based design similar to a web browser. In this case, the user can have several tabs opened from the same PWA but all tied together in the same operating system window, as you can see in the following image:
+
+<video autoplay="" loop="" muted=""><source src="https://storage.googleapis.com/web-dev-uploads/video/8WbTDNrhLsU0El80frMBGE4eMCD3/JwN0btyXFGiT9oPFh2qJ.webm" type="video/webm"></video>
+You can read more about this experimental capability at (Tabbed application mode for PWA [/tabbed-application-mode/]
```
Comment (rachelandrew) [inline src/site/content/en/learn/pwa/windows/index.md:149]:
```suggestion
You can read more about this experimental capability at [Tabbed application mode for PWA](/tabbed-application-mode/).
```

File: src/site/content/en/learn/pwa/windows/index.md [hunk 8]
```diff
@@ -1,11 +1,252 @@
...
+We've mentioned that you can change the window's title by defining the value of the `<title>` element or the `document.title` property. But it's always a string value. What if we could design the title bar as we wish, with HTML, CSS, and images?
+That's where Windows Control Overlay comes in, a new experimental capability in Microsoft Edge and Google Chrome for desktop PWAs.
+
+{% Aside %}
+Tabbed mode and windows control overlay capabilities define new values for the manifest's `display` member. To create compatibility with all devices, the manifest group has created a way to replace the [fallback `display` chain](/learn/pwa/app-design/#display-modes), so you can specify what `display` value to use if your first option is not available. Read more about it at [Preparing for the display modes of tomorrow](/display-override/).
+{% endAside %}
+
+You can read more about this capability at [Customize the window controls overlay of your PWA's title bar](/window-controls-overlay/).
+
+{% Img src="image/RK2djpBgopg9kzCyJbUSjhEGmnw1/9azCEwKziNBBxrTOn3pn.avif", alt="With windows control overlay, you can render content in the title bar", width="800", height="544" %}
```
Comment (rachelandrew) [inline src/site/content/en/learn/pwa/windows/index.md:166]: here's the avif

Comment (rachelandrew) [inline src/site/content/en/learn/pwa/windows/index.md:166]:
```suggestion
{% Img src="image/RK2djpBgopg9kzCyJbUSjhEGmnw1/9azCEwKziNBBxrTOn3pn.avif", alt="With windows control overlay, you can render content in the title bar.", width="800", height="544" %}
```

**Review by rachelandrew — CHANGES_REQUESTED** [review]
A few little fixes and that offending image file.

**Comment by jpmedley** [comment]
I don't know where the images are located.

**Comment by rachelandrew** [comment]
@jpmedley I downloaded it and made a PNG:

{% Img src="image/kheDArv5csY6rvQUJDbWRscckLr1/B5UDAJzKTqCRdSy6B5is.png", alt="ALT_TEXT_HERE", width="800", height="544" %}

File: src/site/content/en/learn/pwa/windows/index.md (hunk 8 above)
Comment (jpmedley) [reply src/site/content/en/learn/pwa/windows/index.md]: I have no way to get that.

File: src/site/content/en/learn/pwa/windows/index.md [hunk 9]
```diff
@@ -1,11 +1,251 @@
...
+We've mentioned that you can change the window's title by defining the value of the `<title>` element or the `document.title` property. But it's always a string value. What if we could design the title bar as we wish, with HTML, CSS, and images?
+That's where Windows Control Overlay comes in, a new experimental capability in Microsoft Edge and Google Chrome for desktop PWAs.
+
+{% Aside %}
+Tabbed mode and windows control overlay capabilities define new values for the manifest's `display` member. To create compatibility with all devices, the manifest group has created a way to replace the [fallback `display` chain](/learn/pwa/app-design/#display-modes), so you can specify what `display` value to use if your first option is not available. Read more about it at [Preparing for the display modes of tomorrow](/display-override/).
+{% endAside %}
+
+You can read more about this capability at [Customize the window controls overlay of your PWA's title bar](/window-controls-overlay/).
+
+{% Img src="image/RK2djpBgopg9kzCyJbUSjhEGmnw1/9azCEwKziNBBxrTOn3pn.avif", alt="With windows control overlay, you can render content in the title bar.", width="800", height="544" %}
```
Comment (rachelandrew) [inline src/site/content/en/learn/pwa/windows/index.md:165]:
```suggestion
{% Img src="image/kheDArv5csY6rvQUJDbWRscckLr1/B5UDAJzKTqCRdSy6B5is.png", alt="With windows control overlay, you can render content in the title bar.", width="800", height="544" %}
```

File: src/site/content/en/learn/pwa/windows/index.md [hunk 10]
```diff
@@ -1,11 +1,251 @@
...
+On desktop, users can open more than one window of the same PWA. Each window will be a different navigation to the same `start_url`, as if you were opening two browser tabs of the same URL.
+From the menu in the PWA, users can select File then New window, and from your PWA code, you can open a new window with the `open()` function. Check the [documentation](https://developer.mozilla.org/docs/Web/API/Window/open) for details.
+
+```js
+function openNewWindow() {
+	window.open("/", "new-window", "width=600,height=600");
+}
+```
+{% Img src="image/RK2djpBgopg9kzCyJbUSjhEGmnw1/CURym4RD66CqcTw4iPl5.png", alt="The same installed PWA with several windows opened on a desktop operating system.", width="800", height="580" %}
+Calling `open()` within a PWA window on iOS or iPadOS returns `null` and doesn't open a window. Opening new windows on Android creates a new in-app browser for the URL—even if the URL is within the scope of your PWA—that typically doesn't trigger an external browsing experience.
```
Comment (rachelandrew) [inline src/site/content/en/learn/pwa/windows/index.md:114]:
```suggestion

Calling `open()` within a PWA window on iOS or iPadOS returns `null` and doesn't open a window. Opening new windows on Android creates a new in-app browser for the URL—even if the URL is within the scope of your PWA—that typically doesn't trigger an external browsing experience.
```

File: src/site/content/en/learn/pwa/windows/index.md (hunk 8 above)
Comment (rachelandrew) [reply src/site/content/en/learn/pwa/windows/index.md]: I fixed it :)

**Review by rachelandrew — APPROVED** [review]

**Comment by rachelandrew** [comment]
@jpmedley and this one is ready too - thank you!

**Pull request merged**